Assignments.java

```diff
@@ -0,0 +1,143 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class Assignments {
+
+	public static void main(String[] args) {
+		Assignment1 assignment = new Assignment1();
+
+		System.out.println(assignment.q1Permutation("Apple", "Pabble"));
+
+		Node<Integer> n1 = new Node<>(7);
+		Node<Integer> n2 = new Node<>(6, n1);
+		Node<Integer> n3 = new Node<>(5, n2);
+		Node<Integer> n4 = new Node<>(4, n3);
+		Node<Integer> n5 = new Node<>(3, n4);
+		Node<Integer> n6 = new Node<>(2, n5);
+		Node<Integer> n7 = new Node<>(1, n6);
+
+		System.out.println(assignment.q2KthToLast(n7, 5));
+	}
+
+	static class Assignment1 {

```

Comment:
It is a good practice to keep classes as small and focused as possible. In this case there are two unrelated (other than by the fact that they are part of a single assignment) things living in a single class - string permutations and operations on lists.

Could you please extract these into two separate classes, each in its own .java file and with its own main() method?

If you would like to group them somehow to make it clear that they are part of the same assignment, an option would be to put them in a single folder (which could also be treated as Java package, but that is completely optional).